service/src/main/java/bio/terra/pipelines/db/JobsDao.java

```diff
@@ -0,0 +1,162 @@
+public class JobsDao {
+  private static final RowMapper<DbJob> DB_JOB_ROW_MAPPER =
+      (rs, rowNum) ->
+          new DbJob(
+              UUID.fromString(rs.getString("job_id")),
+              rs.getString("user_id"),
+              rs.getString("pipeline_id"),
+              rs.getString("pipeline_version"),
+              rs.getTimestamp("time_submitted"),
+              Optional.ofNullable(rs.getTimestamp("time_completed")),
+              rs.getString("status"));
+
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+  private final Logger logger = LoggerFactory.getLogger(JobsDao.class);
+
+  @Autowired
+  public JobsDao(TspsDatabaseConfiguration tspsDatabaseConfiguration) {
+    this.jdbcTemplate = new NamedParameterJdbcTemplate(tspsDatabaseConfiguration.getDataSource());
+  }
+
+  /**
+   * Writes a job to DB. Returns job UUID on success.
+   *
+   * @param job all properties of the job to create
+   * @return jobUuid
+   */
+  @WriteTransaction
+  public UUID createJob(Job job) {
+    final String sql =
+        """
+                        INSERT INTO jobs (job_id, user_id, pipeline_id, pipeline_version, time_submitted, status)
+                        VALUES (:jobId, :userId, :pipelineId, :pipelineVersion, :timeSubmitted, :status)
+                        """;
+
+    final UUID jobUuid = job.getJobId();
+
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("jobId", jobUuid.toString(), VARCHAR)
+            .addValue("userId", job.getUserId(), VARCHAR)
+            .addValue("pipelineId", job.getPipelineId(), VARCHAR)
+            .addValue("pipelineVersion", job.getPipelineVersion(), VARCHAR)
+            .addValue("timeSubmitted", job.getTimeSubmitted(), TIMESTAMP)
+            .addValue("status", job.getStatus(), VARCHAR);
+    try {
+      jdbcTemplate.update(sql, params);
+      logger.info("Inserted record for job {}", jobUuid);
+    } catch (DuplicateKeyException e) {
+      String message = e.getMessage();
+      if (message != null
+          && message.contains("duplicate key value violates unique constraint \"jobs_pkey\"")) {
+        // Job with job_id already exists.
+        throw new DuplicateObjectException(
+            String.format(
+                "Job with id %s already exists - pipelineId %s submitted on %s",
+                jobUuid, job.getPipelineId(), job.getTimeSubmitted()),
+            e);
+      } else {
+        throw e;
+      }
+    }
+    return jobUuid;
+  }
+
+  /**
+   * Return specified job submitted by the user
+   *
+   * @param userId unique identifier of the calling user
+   * @param pipelineId identifier of the pipeline
+   * @param jobId UUID identifier of the job
+   * @return Job object
+   */
+  public Job getJob(String userId, String pipelineId, String jobId) {
+    DbJob dbJob =
+        getDbJobIfExists(userId, pipelineId, jobId)
+            .orElseThrow(() -> new JobNotFoundException(String.format("Job {} not found.", jobId)));
```

Comment:
functionally check that this is working

Reply:
it works! 
`{
  "message": "Job {} not found.",
  "statusCode": 404,
  "causes": []
}
`

Reply:
fixed
`{
  "message": "Job foo not found.",
  "statusCode": 404,
  "causes": []
}`